From d9879bc255741c6a03c5509ee339d31029f9cf9f Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Fri, 21 Nov 2014 17:02:15 +0100
Subject: [PATCH] selinux-access: fix broken ternary operator

it seems to be a typo introduced by ebcf1f97de4f6b1580ae55eb56b1a3939fe6b602
-                _r = selinux_access_check(_b, _m, _u->source_path ?:_u->fragment_path, (permission), &_error); \
+        ({ Unit *_unit = (unit); selinux_generic_access_check(bus,message, _unit->fragment_path ?: _unit->fragment_path, permission,error); })

(cherry picked from commit 1e648011b20c8126412d3cf2699d575d9ba9e0fe)
---
 src/core/selinux-access.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/selinux-access.h b/src/core/selinux-access.h
index bccf0d2913..dd1e8bb9d0 100644
--- a/src/core/selinux-access.h
+++ b/src/core/selinux-access.h
@@ -40,7 +40,7 @@ int mac_selinux_unit_access_check_strv(char **units, sd_bus_message *message, Ma
 #define mac_selinux_unit_access_check(unit, message, permission, error) \
         ({                                                              \
                 Unit *_unit = (unit);                                   \
-                mac_selinux_generic_access_check((message), _unit->fragment_path ?: _unit->fragment_path, (permission), (error)); \
+                mac_selinux_generic_access_check((message), _unit->source_path ?: _unit->fragment_path, (permission), (error)); \
         })
 
 #else
